rpc/handlers.go

```diff
@@ -1343,6 +1348,56 @@ func (h *Handler) SpecVersion() (string, *jsonrpc.Error) {
 	return "v0.5.0", nil
 }
 
+func (h *Handler) SubscribeNewHeads(ctx context.Context) (uint64, *jsonrpc.Error) {
+	w, ok := jsonrpc.ConnFromContext(ctx)
+	if !ok {
+		// If the transport does not give handlers access to the connection, we assume
+		// the method is not supported on the transport.
+		return 0, jsonrpc.Err(jsonrpc.MethodNotFound, nil)
+	}
+	var id uint64
+	id = h.registry.Add(ctx, event.NewSubscription(func(quit <-chan struct{}) error {
+		defer func() {
+			_ = h.registry.Delete(id)
+		}()
+		hexID := "0x" + strconv.FormatUint(id, 16)
+
+		const bufferSize = 2048
```

Comment:
this is way too big, do we need a buffered channel?

Reply:
I agree that's too large. It's good to have a small buffer so we don't drop responses if one takes a while to send.

Reply:
> I made it buffered so we don't drop responses if one takes a while to send

Why would we drop responses?

Reply:
We won't, my bad :man_facepalming:

Reply:
There is this in the `event.*FeedOf.Subscribe` documentation:

> The channel should have ample buffer space to avoid blocking other subscribers.

But we can save memory for now and benchmark to see if blocking is a problem.